### PR TITLE
[repo] Update stale.yml - reduce days before stale to 600

### PR DESCRIPTION
**Pull request by TimothyMothra**
Follow up to https://github.com/open-telemetry/opentelemetry-dotnet/issues/5776

This PR moves the days-before-issue-stale from 900 to 600.
The ultimate goal is to get this down to 300 days.

## Counts of issues

- Currently 309 open issues (Sept 23, 2024).
- older than 300 days (Nov 28, 2023) = 136 issues
- older than 600 days (Feb 1, 2023) = 57 issues ([link](https://github.com/open-telemetry/opentelemetry-dotnet/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-asc+updated%3A%3C%3D2023-02-01+))

## Screenshot of affected issues

<details>
  <summary>Click here to expand</summary>
  
image

image

image

</details>

### CODE REVIEW

**Comment by reyang** [comment]
Could you paste the list of the issues that will be closed here? Having a snapshot would be helpful.

**Comment by TimothyMothra** [comment]
> Could you paste the list of the issues that will be closed here? Having a snapshot would be helpful.

Here's that link: https://github.com/open-telemetry/opentelemetry-dotnet/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-asc+updated%3A%3C%3D2023-02-01+
I also updated my description above.

Just to be clear, nothing will be immediately closed when this PR merges. If this merged today, when the stale automation runs tonight it will tag issues as stale. Those issues would have 7 days for our team to review before they get closed.

**Comment by reyang** [comment]
Not sure if the link would work in the future. I think having a snapshot list would be helpful.

**Comment by TimothyMothra** [comment]
> Not sure if the link would work in the future. I think having a snapshot list would be helpful.

Can you help me understand what's the goal you're trying to accomplish? 
Is this to create a historical record of the issues that would be affected?

I added a screenshot of the affected issues but I want to make sure I'm addressing your concern.

Creating a written list wouldn't be a good use of my time, and worse if I included a list of the hyperlinks that reference would count as activity and would reset the clock.

**Comment by reyang** [comment]
> > Not sure if the link would work in the future. I think having a snapshot list would be helpful.
> 
> Can you help me understand what's the goal you're trying to accomplish? Is this to create a historical record of the issues that would be affected?

I asked this specifically because there are tracking issues that intend to stay open, I think we should review the list instead of blindly close them all. For example, #2622.

**Comment by TimothyMothra** [comment]
> I asked this specifically because there are tracking issues that intend to stay open, I think we should review the list instead of blindly close them all. 

Ahh makes sense. When an issue gets marked stale, we're really depending on the original person to comment that this is still needed. Worst case if something gets closed, we would gladly reopen it. :)

I think you missed the first PR when we started this effort. We started at 1200 days old, so we could individually review a small group of issues at a time without overwhelming maintainers. We absolutely do not want to hastily close issues!

We've also discussed excluding certain labels but are currently using the "Future" milestone to exclude issues from Stale.

**Review by Kielek — APPROVED** [review]

**Comment by rajkumar-rangaraj** [comment]
@TimothyMothra If you could reopen https://github.com/open-telemetry/opentelemetry-dotnet/issues/5776 and explain the process we follow for issues marked as stale, it would be helpful. Once this exercise is complete, we can work on formalizing the process and documenting it in the repository

**Review by rajkumar-rangaraj — APPROVED** [review]

**Comment by rajkumar-rangaraj** [comment]
> Not sure if the link would work in the future. I think having a snapshot list would be helpful.

I like this idea. It will also help us track how many issues were closed as stale and how many we reviewed and moved forward.

**Pull request merged**
